packages/fluentui/react-component-ref/src/Ref.tsx

```diff
@@ -1,16 +1,159 @@
 import * as React from 'react';
 import * as ReactIs from 'react-is';
+import * as ReactDOM from 'react-dom';
 
-import { RefFindNode } from './RefFindNode';
-import { RefForward } from './RefForward';
-import { RefProps } from './utils';
+import { handleRef } from './utils';
 
-export const Ref: React.FunctionComponent<RefProps> = props => {
-  const { children, innerRef, ...rest } = props;
+export interface RefProps {
+  children: React.ReactElement;
 
-  const child = React.Children.only(children);
-  const ElementType = ReactIs.isForwardRef(child) ? RefForward : RefFindNode;
-  const childWithProps = child && rest && Object.keys(rest).length > 0 ? React.cloneElement(child, rest) : child;
+  /**
+   * Called when a child component will be mounted or updated.
+   *
+   * @param node - Referred node.
+   */
+  innerRef: React.Ref<HTMLElement>;
+}
 
-  return <ElementType innerRef={innerRef}>{childWithProps}</ElementType>;
+export interface RefState {
+  kind: 'self' | 'forward' | 'find' | null;
+}
+
+// ========================================================
+// react/packages/react-reconciler/src/ReactFiber.js
+// ========================================================
+
+type Fiber = {
+  // Tag identifying the type of fiber.
+  tag: string;
+  // The resolved function/class/ associated with this fiber.
+  type: any;
 };
+
+/**
+ * Detects if a passed element is a Fiber object instead of an element. Is needed as `ReactDOM.findDOMNode()` returns
+ * a Fiber in `react-test-renderer` that can cause issues with tests. Is used only in non-production env.
+ *
+ * @see https://github.com/facebook/react/issues/7371#issuecomment-317396864
+ * @see https://github.com/Semantic-Org/Semantic-UI-React/issues/4061#issuecomment-694895617
+ */
+function isFiberRef(node: Element | Fiber | Text | null): boolean {
+  if (node === null) {
+    return false;
+  }
+
+  if (node instanceof Element || node instanceof Text) {
```

Comment:
in the context of Teams is it safe to use `instanceof` ? https://codesandbox.io/s/angry-poincare-64npnn?file=/example.tsx

Reply:
Well, it was there for ages and no-one complained yet... Especially taking into account that this will not go through iframes as it's only used in `findDOMNode()` case.